src/index.ts

```diff
@@ -59,7 +60,8 @@ export const start = () => calculator?.start(performance.now());
  * @dropbox/ttvc that a user interaction has occurred and continuing the
  * measurement may produce an invalid result.
  */
-export const cancel = () => calculator?.cancel();
+export const cancel = (reason: CancellationReason = CancellationReason.USER_INTERACTION) =>
```

Comment:
My original thinking on this is that the API would look something like `.cancelOnUserInteraction(.USER_CLICK | .USER_HOVER | ...)`
The reason being that this forces a set of valid user interactions (and maybe we even somehow make it such that click is not a part of the public API, but only the TTVC library it self can emit it). So if someone wants to use it, they have to select from a very specific set of reasons. That combined with a `onCancel` callback would allow us to collect metrics on the very specific types of cancellations.

Reply:
One issue with that plan is that TypeScript types aren't actually enforced if consumed in a regular JS context.  i.e. We can't actually enforce that nobody passes an arbitrary string unless we're willing to make this `cancel` API throw at runtime.

I'm also skeptical that, sitting here, we can correctly guess all of the possible user interactions that an app might want to differentiate.